lib/matplotlib/lines.py

```diff
@@ -898,6 +925,8 @@ def set_color(self, color):
 
         ACCEPTS: any matplotlib color
         """
+        if color != self._color:
```

Comment:
When I started this I was trying to be clever and short-circuit the stale logic in cases where there is no change and then gave up an that made made calling `set_*` to always mark the artist (and hence up the chain) as stale.

In this case, even if they are the same color with different reps, marking as stale is not _really_ a problem.